### PR TITLE
Best practices

### PR DESCRIPTION
**Pull request by medhamsh**
- Checked for the best practices as per the doc
- Being Java as the dep, we need to migrate it to the ind9-ops soon

@arijit-indix @ashwanthkumar @gokul-ps

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  - ANSIBLE_VERSION=2.0.0.2
+  - ANSIBLE_VERSION=2.0.0.1
+  - ANSIBLE_VERSION=2.0.0.0
+
+branches:
+  only:
+    - master
+
+matrix:
+  allow_failures:
```
Comment (ashwanthkumar) [inline .travis.yml:22]: Since we don't test against the 1.9.1 version. I guess we can remove this section.

File: defaults/main.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+---
+gocd_common_user: go
+gocd_common_userid: 5000
+gocd_common_group: go
+gocd_common_groupid: 5000
+gocd_common_debian_repository: http://dl.bintray.com/gocd/gocd-deb/
```
Comment (ashwanthkumar) [inline defaults/main.yml]: Please don't use bintray, GoCD guys has moved away from that. 
Ref - https://groups.google.com/d/msg/go-cd/nadSZDN4eM0/2OwJwvygBwAJ

File: files/thoughtworks-go-download.repo [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# {{ansible_managed}}
+
+[thoughtworks-go-download]
+name=Thoughtworks Go Repository
+baseurl=http://dl.bintray.com/gocd/gocd-rpm/
```
Comment (ashwanthkumar) [inline files/thoughtworks-go-download.repo]: Same goes here - Please use the download.go.cd.

File: defaults/main.yml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+---
+gocd_common_user: go
+gocd_common_userid: 5000
```
Comment (ashwanthkumar) [inline defaults/main.yml]: I don't see this and `gocd_common_groupid` being used. Can we remove this? Also please go through all the existing variables and remove the one's we don't use anywhere within the role.

**Comment by ashwanthkumar** [comment]
@medhamsh The CI seems to be broken, we should fix that as well. 

Please address the comments above, rest LGTM.

File: .travis.yml (hunk 1 above)
Comment (arijit-indix) [inline .travis.yml:22]: > Since we don't test against the 1.9.1 version. I guess we can remove this section.

👍

File: defaults/main.yml [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+gocd_common_group: go
+gocd_common_groupid: 5000
+gocd_common_debian_repository: http://dl.bintray.com/gocd/gocd-deb/
+gocd_common_configure_ssh: false
+gocd_common_ssh_known_domain: github.com
+gocd_common_java_versions:
+  - oracle-java8-installer
+  - oracle-java8-set-default
```
Comment (arijit-indix) [inline defaults/main.yml:9]: Any package (preferably along with its dependencies) need to have specific version pinned.

File: tasks/main.yml [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+  when: gocd_common_configure_ssh
+
+- name: SSH private key
+  copy:
+    dest: /var/go/.ssh/{{gocd_common_ssh_private_key | basename}}
+    src: "{{gocd_common_ssh_private_key}}"
+    group: "{{gocd_common_group}}"
+    mode: 0600
+    owner: "{{gocd_common_user}}"
+  when: gocd_common_configure_ssh
```
Comment (arijit-indix) [inline tasks/main.yml:60]: Where is the private key retrieved from ? It should be in the S3 bucket and downloaded using Iam role.

**Pull request closed without merging**